src/main/java/io/nats/client/api/StreamInfoOptions.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.client.api;
+
+import io.nats.client.support.JsonSerializable;
+
+import static io.nats.client.support.ApiConstants.DELETED_DETAILS;
+import static io.nats.client.support.ApiConstants.SUBJECTS_FILTER;
+import static io.nats.client.support.JsonUtils.*;
+
+/**
+ * Object used to make a request for special stream info requests
+ */
+public class StreamInfoOptions implements JsonSerializable {
+    private final String subjectsFilter;
+    private final boolean deletedDetails;
+
+    private StreamInfoOptions(String subjectsFilter, boolean deletedDetails) {
+        this.subjectsFilter = subjectsFilter;
+        this.deletedDetails = deletedDetails;
+    }
+
+    /**
+     * Create options that get subject information, filtering for subjects. Wildcards are allowed.
+     * @param subjectsFilter the subject filter. &gt; is equivalent to all
+     * @return the StreamInfoOptions object
+     */
+    public static StreamInfoOptions filterSubjects(String subjectsFilter) {
+        return new Builder().filterSubjects(subjectsFilter).build();
+    }
+
+    /**
+     * Create options that get subject information, filtering for all subjects.
+     * @return the StreamInfoOptions object
+     */
+    public static StreamInfoOptions allSubjects() {
+        return new Builder().allSubjects().build();
+    }
+
+    /**
+     * Create options that get deleted details.
+     * @return the StreamInfoOptions object
+     */
+    public static StreamInfoOptions deletedDetails() {
+        return new Builder().deletedDetails().build();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public String toJson() {
+        StringBuilder sb = beginJson();
+        addField(sb, SUBJECTS_FILTER, subjectsFilter);
```

Comment:
This JSON field should be added ONLY when the subject filter is actually specified. Is addField() doing that under the cover?

Reply:
yes, addField checks for null/empty and doesn't add it.